File: Practice/Topics/Greedy/2127B.py
import sys
import math

sys.stdin = open("input.txt","r")
sys.stdout = open("output.txt","w")

def solve():
	n,x = map(int,input().split())

	s = input()

	distleft = 0
	distright = 0
	lefthash = 0
	leftnearest = 1e9
	for i in range(x-1,-1,-1):
		if s[i] == '#':
			lefthash += 1
			if leftnearest == 1e9:
				leftnearest = i
		distleft += 1
	righthash = 0

	rightnearest = -1
	for i in range(x-1,n):
		if (s[i] == '#'):
			righthash += 1
			if rightnearest == -1:
				rightnearest = i
		distright += 1
	if (lefthash == 0 and righthash == 0):
		print('1')
		return
	
	if (x - leftnearest < rightnearest - x):
		print(distleft)
	else : print(distright)


tc = 1
tc = int(input())
for _ in range(tc):
	solve()

File: Practice/Topics/Greedy/2127B.cpp
#include<bits/stdc++.h>
using namespace std;
#define inf 1e9
void solve(){
	int n,x;
	cin >> n >> x;
	string s;
	cin >> s;

	x --;

	if (x == 0 || x == n-1){
		cout << 1<< endl;
		return;
	}

	int lw = -inf;
	int rw = inf;
	for (int i = x -1;i >= 0 ; i--){
		if (s[i] == '#'){
			lw = i;
			break;
		}
	}

	for (int i = x + 1;i < n ;i++){
		if (s[i] == '#'){
			rw = i;
			break;
		}
	}

	if (lw == -inf && rw == inf){
		cout << 1 << endl;
		return;
	}

	cout << max(min(x+1,n - rw + 1),min(lw + 2, n -x))<<"\n";

}

int main(){
	#ifndef ONLINE_JUDGE
	freopen("input.txt","r",stdin);
	freopen("output.txt","w",stdout);
	#endif
	int tc = 1;
	cin >> tc;

	while (tc--){
		solve();
	}
}